ompi/mca/coll/base/coll_base_alltoallv.c

```diff
@@ -285,11 +285,12 @@ ompi_coll_base_alltoallv_intra_basic_linear(const void *sbuf, const int *scounts
      * i.e., by the end of this call, all the requests are free-able.
      * So free them anyway -- even if there was an error, and return the
      * error after we free everything. */
-    err = ompi_request_wait_all(nreqs, data->mcct_reqs,
-                                MPI_STATUSES_IGNORE);
-
-    /* Free the requests. */
-    ompi_coll_base_free_reqs(data->mcct_reqs, nreqs);
+    err = ompi_request_wait_all(nreqs, reqs, MPI_STATUSES_IGNORE);
+    if( MPI_SUCCESS == err )
+        return MPI_SUCCESS;
```

Comment:
@bosilca 
Do not agree: these are persistent requests and should be freed in all cases
==> don't return